evalml/model_understanding/graphs.py

```diff
@@ -690,14 +690,24 @@ def partial_dependence(
 
     _raise_value_error_if_mostly_one_value(feature_list, percentiles[1])
     wrapped = evalml.pipelines.components.utils.scikit_learn_wrapped_estimator(pipeline)
-    preds = sk_partial_dependence(
-        wrapped,
-        X=X,
-        features=features,
-        percentiles=percentiles,
-        grid_resolution=grid_resolution,
-        kind=kind,
-    )
+    try:
+        preds = sk_partial_dependence(
+            wrapped,
+            X=X,
+            features=features,
+            percentiles=percentiles,
+            grid_resolution=grid_resolution,
+            kind=kind,
+        )
+    except ValueError as e:
```

Comment:
Thoughts on us doing the check rather than relying on sklearn? My reasoning is that with @bchen1116 's #2454 , we're going to be computing the grid ourselves for datetimes and I think we'd want to have this same check in place for dates as well?

Maybe the right thing to do is modify #2454 to compute the grid ourselves for all features rather than having separate branches for datetimes vs other features. That may make it easier to put the "can we compute a valid grid for these features?" checks in one place? Curious what you think!

Reply:
My first crack was reproducing the check myself, rather than just let sklearn's partial dependence fail.  I decided against that, ultimately, as I didn't want to take the risk that sklearn changes the current criteria behind the check and I wanted the catch to be a little more resilient to that.

I'm not even sure how the same check that's failing within partial dependence would handle date times.  I guess you could define a relative tolerance of 1E-5 and it mean something if you convert a datetime to like linux time or something.  

I'm also not against rejecting this PR if you don't think catching the existing sklearn exception is still the right move.  If we're more in favor of computing, for sure, a functional grid, then we can go that route and I can flush this sucker down the toilet.